packages/strapi-plugin-graphql/services/GraphQL.js

```diff
@@ -17,6 +17,424 @@ const GraphQLJSON = require('graphql-type-json');
+   *     username: [UserConnectionUsername]
+   *     age: [UserConnectionAge]
+   *  }
+   * 
+   *  type UserConnectionUsername {
+   *    key: String
+   *    connection: UserConnection
+   *  }
+   * 
+   *  type UserConnectionAge {
+   *    key: Int
+   *    connection: UserConnection
+   *  }
+   *
+   */
+  formatModelConnectionsGQL: function(fields, model, name, modelResolver) {
+    const { globalId } = model;
+
+    const connectionGlobalId = `${globalId}Connection`;
+    const aggregatorFormat = this.formatConnectionAggregator(fields, model, name);
+    const groupByFormat = this.formatConnectionGroupBy(fields, model, name);
+    const connectionFields = {
+      values: `[${globalId}]`,
+      groupBy: `${globalId}GroupBy`,
+      aggregate: `${globalId}Aggregator`,
+    };
+    
+    let modelConnectionTypes = `type ${connectionGlobalId} {${this.formatGQL(connectionFields)}}\n\n`;
+    if (aggregatorFormat) {
+      modelConnectionTypes += aggregatorFormat.type;
+    }
+    modelConnectionTypes += groupByFormat.type;
+
+    return {
+      globalId: connectionGlobalId,
+      type: modelConnectionTypes,
+      query: {
+        [`${pluralize.plural(name)}Connection(sort: String, limit: Int, start: Int, where: JSON)`]: connectionGlobalId,
+      },
+      resolver: {
+        Query: {
+          [`${pluralize.plural(name)}Connection`]: (obj, options, context) => { // eslint-disable-line no-unused-vars
+            const params = Object.assign(
+              {},
+              this.convertToParams(_.omit(options, 'where')),
+              options.where
+            );
+            return strapi.utils.models.convertParams(name, params);
+          }
+        },
+        [connectionGlobalId]: {
+          values: (obj, option, context) => {
+            // Object here contains the key/value of the field that has been grouped-by
+            // for instance obj = { where: { country: 'USA' } } so the values here needs to be filtered according to the parent value
+            return modelResolver(obj, obj, context);
+          },
+          groupBy: (obj, option, context) => { // eslint-disable-line no-unused-vars
+            // There is noting to resolve here, it's the aggregation resolver that will take care of it
+            return obj;
+          },
+          aggregate: (obj, option, context) => { // eslint-disable-line no-unused-vars
+            // There is noting to resolve here, it's the aggregation resolver that will take care of it
+            return obj;
+          },
+        },
+        ...aggregatorFormat.resolver,
+        ...groupByFormat.resolver,
+      },
+    };
+  },
+
+  /**
+   * Generate the connection type of each non-array field of the model
+   * 
+   * @return {String}
+   */
+  generateConnectionFieldsTypes: function (fields, model) {
+    const { globalId } = model;
+
+    const primitiveFields = this.getFieldsByTypes(
+      fields,
+      this.isNotOfTypeArray,
+      type => this.extractType(type),
```

Comment:
Do we need this method and also `isNotPrimitive`, `isNotOfTypeArray`? We have access to the models `strapi.models` and the associations related to it `strapi.models.xxx.associations`. 

PS: My comment might be not very relevant (I don't have the full logic in mind) but it's just a remember to avoid the creation of unnecessaries functions 🙈

Reply:
I'd love to do it, but the model attributes includes the non-transformed field types so if I do so I'll have to duplicate some of the logic, I think it's better to do like this